src/services/admin/AuditLogsService.ts

```diff
@@ -0,0 +1,500 @@
+    )
+
+    return ResultAsync.fromPromise(
+      octokit.paginate(octokit.repos.listCommits, {
+        owner: "isomerpages",
+        repo: sessionData.siteName,
+        since: sinceDate,
+        until: untilDate,
+        per_page: 100,
+      }),
+      (error) => {
+        logger.error(
+          `Error occurred when getting the list of commits for the site ${
+            sessionData.siteName
+          } from GitHub: ${JSON.stringify(error)}`
+        )
+
+        return new AuditLogsError(
+          `Error occurred when getting the list of commits for the site ${sessionData.siteName}`
+        )
+      }
+    )
+      .andThen((commits) =>
+        ResultAsync.combine(
+          commits
+            .filter(
+              (commit) =>
+                // Note: We can ignore merge commits since we are on the staging branch
+                !commit.commit.message.startsWith("Merge ")
+            )
+            .map((commit) => {
+              const { message } = commit.commit
+
+              if (message.startsWith("{")) {
+                try {
+                  const parsedMessage: IsomerCommitMessage = JSON.parse(message)
+
+                  return this.getAuditLogActorNameFromId(parsedMessage.userId)
+                    .orElse((error) => {
+                      if (typeof error === "boolean") {
+                        return okAsync(
+                          commit.commit.author?.name || "Unknown author"
+                        )
+                      }
+
+                      return errAsync(error)
+                    })
+                    .andThen((actorEmail) =>
+                      okAsync({
+                        timestamp: new Date(commit.commit.author?.date || ""),
+                        activity: AuditableActivity.SavedChanges,
+                        actor: actorEmail,
+                        page: parsedMessage.fileName || "",
+                        remarks: parsedMessage.message,
+                      })
+                    )
+                } catch (error) {
+                  logger.error(
+                    `Error parsing JSON in commit ${commit.sha} from ${sessionData.siteName}: ${error}\n`
+                  )
+                  return errAsync(
+                    new AuditLogsError("Error parsing JSON in commit")
+                  )
+                }
+              }
+
+              return this.getAuditLogActorNameFromGitHubId(
+                commit.commit.author?.name ?? ""
+              )
+                .orElse((error) => {
+                  if (typeof error === "boolean") {
+                    return okAsync(
+                      commit.commit.author?.name || "Unknown author"
+                    )
+                  }
+
+                  return errAsync(error)
+                })
+                .andThen((actorEmail) =>
+                  okAsync({
+                    timestamp: new Date(commit.commit.author?.date || ""),
```

Comment:
hmm this makes it to today's timestamp, rather than idk the timestamp... is this intended?

Reply:
This is more to satisfy the types, because every commit will have a timestamp.

Reply:
arh ok